pkgs/build-support/kernel/modules-closure.sh

```diff
@@ -66,8 +66,8 @@ for module in $rootModules; do
     fi
 done
 
-mkdir -p $out/lib/firmware
```

Comment:
For reference, this is the error I get:
```
Error: failed to get symlink metadata for "/nix/store/85adj5c0zgrn6mi8qs8i3mxvmsipkid4-linux-armv6l-unknown-linux-gnueabihf-6.1.21-1.20230405-modules-shrunk/lib/firmware"

Caused by:
    No such file or directory (os error 2)

Location:
    src/main.rs:176:22
```

Reply:
We can put this line back (with a comment) if you want.  There's no harm having an empty firmware directory.

Reply:
Yes, it is either that or change the initrd to use:
```
"/lib".source = "${modulesClosure}/lib"; 
```
That would make it more like what you did for the regular initrd, but I don't if it would break anything:
https://github.com/NixOS/nixpkgs/blob/a0ba4615da4053ce59c6367a3aa276c1a8d24d9e/nixos/modules/system/boot/stage-1.nix#L352-L354

Reply:
Oh yeah, that seems like the thing to do if it works.

Reply:
#273695